homeassistant/components/accuweather/weather.py

```diff
@@ -52,106 +51,127 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Add a AccuWeather weather entity from a config_entry."""
+    accuweather_data: AccuWeatherData = hass.data[DOMAIN][entry.entry_id]
 
-    coordinator: AccuWeatherDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities([AccuWeatherEntity(accuweather_data)])
 
-    async_add_entities([AccuWeatherEntity(coordinator)])
 
-
-class AccuWeatherEntity(
-    SingleCoordinatorWeatherEntity[AccuWeatherDataUpdateCoordinator]
-):
+class AccuWeatherEntity(CoordinatorWeatherEntity):
     """Define an AccuWeather entity."""
 
     _attr_has_entity_name = True
     _attr_name = None
 
-    def __init__(self, coordinator: AccuWeatherDataUpdateCoordinator) -> None:
+    def __init__(self, accuweather_data: AccuWeatherData) -> None:
         """Initialize."""
-        super().__init__(coordinator)
+        super().__init__(
+            observation_coordinator=accuweather_data.coordinator_observation,
+            daily_coordinator=accuweather_data.coordinator_daily_forecast,
+        )
```

Comment:
Huh, how does this constructor work?

Reply:
https://github.com/home-assistant/core/blob/c7e6f3696f16fc3a5626387cec5fac7e29d2b32f/homeassistant/components/weather/__init__.py#L1086-L1114

Reply:
Oh cool :)